stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/logic/channel/internal/ChannelStateLogic.kt

```diff
@@ -186,6 +187,10 @@ internal class ChannelStateLogic(
     override fun upsertMessages(messages: List<Message>, shouldRefreshMessages: Boolean, updateCount: Boolean) {
         when (shouldRefreshMessages) {
             true -> {
+                messages.filter { message -> message.isReply() }
+                    .forEach { message ->
+                        mutableState.addQuotedMessage((message.replyTo?.id ?: message.replyMessageId)!!, message.id)
```

Comment:
I would maybe replace the `!!` with `requireNotNull()`

Something like:

```kotlin
val replyId = requireNotNull((message.replyTo?.id ?: message.replyMessageId)) { "Message filtered as reply doesn't have replyId!" }

mutableState.addQuotedMessage(replyId, message.id)
```

Reply:
I used optional parameters, instead of `requireNotNull`. I believe that in this case not updating what is not there is better then throwing a crash.

Reply:
Well you didn't really use optional because you're using force unwrap after checking the optionals.

So if they are optional/null, you fetch it and then force unwrap. But you're filtering messages which are `isReply()`, so they should have the data, we should throw if somehow the check passes but the data isn't there